api/v1alpha1/biosversionset_types.go

```diff
@@ -0,0 +1,75 @@
+// SPDX-FileCopyrightText: 2025 SAP SE or an SAP affiliate company and IronCore contributors
+// SPDX-License-Identifier: Apache-2.0
+
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// EDIT THIS FILE!  THIS IS SCAFFOLDING FOR YOU TO OWN!
+// NOTE: json tags are required.  Any new fields you add must have json tags for the fields to be serialized.
+
+// BIOSVersionSetSpec defines the desired state of BIOSVersionSet.
+type BIOSVersionSetSpec struct {
+
+	// ServerSelector specifies a label selector to identify the servers that are to be selected.
+	// +required
+	ServerSelector metav1.LabelSelector `json:"serverSelector"`
+
+	// BiosVersionTemplate defines the template for the BIOSversion Resource to be applied to the servers.
+	BiosVersionTemplate VersionUpdateSpec `json:"biosVersionTemplate,omitempty"`
+
+	// ServerMaintenancePolicy is a maintenance policy to be enforced on the server.
+	// +optional
+	ServerMaintenancePolicy ServerMaintenancePolicy `json:"serverMaintenancePolicy,omitempty"`
+}
+
+// BIOSVersionSetStatus defines the observed state of BIOSVersionSet.
+type BIOSVersionSetStatus struct {
+	// TotalServers is the number of server in the set.
+	TotalServers int32 `json:"totalServers,omitempty"`
+	// TotalVersionResource is the number of Settings current created by the set.
+	TotalVersionResource int32 `json:"totalVersionResource,omitempty"`
+	// Pending is the total number of pending server in the set.
+	Pending int32 `json:"pending,omitempty"`
+	// InMaintenance is the total number of server in the set that are currently in InProgress.
+	InProgress int32 `json:"inProgress,omitempty"`
+	// Completed is the total number of completed server in the set.
+	Completed int32 `json:"completed,omitempty"`
+	// Failed is the total number of failed server in the set.
+	Failed int32 `json:"failed,omitempty"`
```

Comment:
Can we change those status fields and have something similar like the `ReplicaSets` in k8s? https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.27/#replicasetstatus-v1-apps